src/realm/object-store/c_api/realm.cpp

```diff
@@ -263,11 +263,33 @@ RLM_API bool realm_compact(realm_t* realm, bool* did_compact)
 {
     return wrap_err([&]() {
         auto& p = **realm;
-        *did_compact = p.compact();
+        if (did_compact)
```

Comment:
This is not related to this PR, however we need to keep this pattern in the whole C API. Unless we don't decide to clean all these checks up (which I would agree on).